samples/gallery/src/main/java/com/bumptech/glide/samples/gallery/HorizontalGalleryFragment.kt

```diff
@@ -1,50 +1,559 @@
+        override fun onStop() {}
+        override fun onDestroy() {}
+
+        override fun onLoadStarted(placeholder: Drawable?) {
+          // More hackery - expect that the given placeholder drawable is the same instance as the
+          // one passed here. If the given placeholder is set as the initial value, then this will
+          // skip emitting the placeholder twice and therefore let us skip a useless recomposition.
+          trySend(placeholder)
+        }
+
+        override fun onLoadFailed(errorDrawable: Drawable?) {
+          trySend(errorDrawable)
+        }
+
+        override fun onResourceReady(resource: Drawable, transition: Transition<in Drawable>?) {
+          trySend(resource)
+        }
+
+        override fun onLoadCleared(placeholder: Drawable?) {
+          trySend(placeholder)
+        }
+
+        override fun setRequest(request: Request?) {
+          currentRequest = request
+        }
+
+        override fun getRequest(): Request? {
+          return currentRequest
+        }
+
+        override fun getSize(cb: SizeReadyCallback) {
+          val localResolvedSize = resolvedSize
+          if (localResolvedSize != null) {
+            cb.onSizeReady(localResolvedSize.width, localResolvedSize.height)
+          } else {
+            sizeReadyCallbacks.add(cb)
+          }
+        }
+
+        override fun removeCallback(cb: SizeReadyCallback) {
+          sizeReadyCallbacks.remove(cb)
         }
       }
+      Log.d("TEST", "Start load for $model into $target")
+      requestBuilder.into(target)
+      awaitClose {
+        Log.d("TEST", "Close $model")
+        requestManager.clear(target)
+      }
     }
   }
 
-  override fun onCreateView(
-    inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?,
-  ): View? {
-    val result = inflater.inflate(R.layout.recycler_view, container, false)
-    recyclerView = result.findViewById<View>(R.id.recycler_view) as RecyclerView
-    val layoutManager = GridLayoutManager(activity, 1)
-    layoutManager.orientation = RecyclerView.HORIZONTAL
-    recyclerView.layoutManager = layoutManager
-    recyclerView.setHasFixedSize(true)
-
-    val glideRequests = GlideApp.with(this)
-    adapter = RecyclerAdapter(requireContext(), glideRequests)
-    val preloader = RecyclerViewPreloader(glideRequests, adapter, adapter, 3)
-    recyclerView.addOnScrollListener(preloader)
-    recyclerView.adapter = adapter
-    return result
+  @Composable
+  private fun rememberGlidePainter(
+    requestManager: RequestManager,
+    model: Any?,
+    requestBuilder: RequestBuilder<Drawable>,
+    size: ResolvableGlideSize,
+  ): GlidePainter =
+    // TODO(judds): Calling onRemembered here manually might make a minor improvement in how quickly
+    //  the image load is started, but it also triggers a recomposition. I can't figure out why it
+    //  triggers a recomposition
+    remember { GlidePainter(requestManager, model, requestBuilder, size) }
```

Comment:
Probably needs some keys, otherwise the `GlidePainter` will be using stale captures if any of the parameters change.
```suggestion
    remember(requestManager, model, requestBuilder, size) {
      GlidePainter(requestManager, model, requestBuilder, size)
    }
```

Reply:
Done.

I'm a little concerned that the way I'm handling the size is incorrect here. The AsyncSize is a data class that wraps a function. The output of the function is dependent on the Modifiers. So in that case we probably should try to find a way to recompose if the input modifier changes?